client/src/AppBundle/Service/Audit/AuditEvents.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace AppBundle\Service\Audit;
+
+final class AuditEvents
+{
+    const CLIENT_DISCHARGED = 'CLIENT_DISCHARGED';
+    const CLIENT_DISCHARGED_ADMIN_TRIGGER = 'ADMIN_BUTTON';
+    const CLIENT_DISCHARGED_CSV_TRIGGER = 'CSV_UPLOAD';
+
+    /**
+     * @param $trigger
+     * @param $caseNumber
+     * @return array
+     */
+    public function clientDischarged(string $trigger, string $caseNumber, string $dischargedBy = null): array
+    {
+        $event = [
+            'trigger' => $trigger,
+            'case_number' => $caseNumber
+        ];
+
+        if (null !== $dischargedBy) {
+            $event['discharged_by'] = $dischargedBy;
```

Comment:
What scenarios would the discharged_by be null? Could leave a gap in audit trail if it's not set. 

Is this to cover an automated process? In which case maybe be explicit in capturing that.

Reply:
Yes it's to cover an automated process, which we are looking at removing in my current ticket, so now in reality we will always set this field. I expect that ticket to go ahead of this one into master, so this PR can go on hold until I update it to not include the automated process.